autogen/agentchat/conversable_agent.py

```diff
@@ -721,14 +721,20 @@ def reset_consecutive_auto_reply_counter(self, sender: Optional[Agent] = None):
         else:
             self._consecutive_auto_reply_counter[sender] = 0
 
-    def clear_history(self, agent: Optional[Agent] = None):
+    def clear_history(self, agent: Optional[Agent] = None, preserve_messages: Optional[int] = None):
```

Comment:
“Num_perserved_messages”

Reply:
@ekzhu  Changed variable name to 'nr_messages_to_preserve'. Seems to be most readable for me. Will be ok?